src/main/java/nextstep/ladder/domain/line/Line.java

```diff
@@ -2,31 +2,34 @@
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Random;
+import nextstep.ladder.domain.RandomGenerator;
 
 public class Line {
 
-    private final Random random = new Random();
-
-    private List<Boolean> points = new ArrayList<>();
+    private List<LadderConstructionStatus> points = new ArrayList<>();
 
     public Line(int countOfPerson) {
-        boolean before = false;
-        for (int i = 0; i < countOfPerson - 1; i++) {
-            before = addLineValue(before);
-        }
+        initializeLine(countOfPerson);
     }
 
-    private boolean addLineValue(boolean before) {
-        if (before == false) {
-            boolean currentState = random.nextBoolean();
-            points.add(currentState);
-            return currentState;
+    private void initializeLine(int countOfPerson) {
+        LadderConstructionStatus ladderConstructionStatus = new LadderConstructionStatus(false);
+        for (int count = 0; count < countOfPerson - 1; count++) {
+            ladderConstructionStatus = addLineValue(ladderConstructionStatus);
         }
-        return points.add(false);
     }
 
-    public List<Boolean> getPoints() {
+    private LadderConstructionStatus addLineValue(
+        LadderConstructionStatus previousLadderConstructionStatus) {
+        LadderConstructionStatus newStatus =
+            previousLadderConstructionStatus.isLadderConstructionStatusFalse()
+                ? new LadderConstructionStatus(RandomGenerator.createRandomBoolean())
+                : new LadderConstructionStatus(false);
+        points.add(newStatus);
```

Comment:
isLadderConstructionStatusFalse() 메서드를 활용해 삼항 연산자를 사용한다면 getter를 활용하는 것과 동일합니다.
메시지를 보내 다음 값을 구하면 어떨까요?
```suggestion
        LadderConstructionStatus newStatus = previousLadderConstructionStatus.next();
        points.add(newStatus);
```

Reply:
엇 훨씬 좋네요
피드백 감사드립니다